pkg/apiserver/patch/patch.go

```diff
@@ -0,0 +1,454 @@
+/*
```

Comment:
I'm not keen on contributing to code sprawl. We currently have code sprinkled all over a very flat directory layout -- #4851, and putting the code into util would make it harder to extract apiserver later #2742. I suggest leaving the code under apiserver until we figure out how we should actually be structuring our codebase.